manifests/repository.pp

```diff
@@ -17,15 +17,22 @@
 #   The major bareos release version which should be used
 # @param password
 #   The major bareos release version which should be used
+# @param https
+#   Wether or not https should be used in repo URL
 #
 class bareos::repository (
   String              $release             = 'latest',
   Optional[String[1]] $gpg_key_fingerprint = undef,
   Boolean             $subscription        = false,
   Optional[String]    $username            = undef,
   Optional[String]    $password            = undef,
+  Boolean             $https               = true,
```

Comment:
This needs a spec test that shows the default case as well as what happens when the parameter is set to `false`.

Reply:
The default should, very unfortunately, stay `http` as this is a breaking change to modify the behavior by changing the protocol. If you are doing a lot of work with this module, you might get feedback about any other wanted breaking changes and coordinate to get those in and a major version bump.